applications/fax/src/fax_smtp.erl

```diff
@@ -670,90 +669,110 @@ process_parts([], #state{filename='undefined'
     {'ok', State#state{errors=[<<"no valid attachment">> | Errors]}};
 process_parts([], State) ->
     {'ok', State};
-process_parts([{Type, SubType, Headers, Parameters, BodyPart}
+process_parts([{Type, SubType, _Headers, Parameters, BodyPart}
                |Parts
               ], State) ->
     {_ , NewState}
-        = process_part(<<Type/binary, "/", SubType/binary>>
-                       ,Headers
+        = maybe_process_part(<<Type/binary, "/", SubType/binary>>
                        ,Parameters
                        ,BodyPart
                        ,State
                       ),
     process_parts(Parts, NewState).
 
--spec process_part(ne_binary(), wh_proplist(), wh_proplist(), binary() | mimemail:mimetuple(), state()) ->
+-spec maybe_process_part(ne_binary(), wh_proplist(), binary() | mimemail:mimetuple(), state()) ->
                           {'ok', state()}.
-process_part(<<"application/pdf">>=CT, _Headers, _Parameters, Body, State) ->
-    lager:debug("part is application/pdf"),
-    Filename = <<"/tmp/email_attachment_", (wh_util:to_binary(wh_util:current_tstamp()))/binary, ".pdf">>,
-    'ok' = write_file(Filename, Body),
-    {'ok', State#state{filename=Filename
-                       ,content_type=CT
-                      }};
-process_part(<<"image/tiff">>=CT, _Headers, _Parameters, Body, State) ->
-    lager:debug("Part is image/tiff"),
-    Filename = <<"/tmp/email_attachment_", (wh_util:to_binary(wh_util:current_tstamp()))/binary, ".tiff">>,
-    'ok' = write_file(Filename, Body),
-    {'ok', State#state{filename=Filename
-                       ,content_type=CT
-                      }};
-process_part(<<"application/octet-stream">>, _Headers, Parameters, Body, State) ->
+maybe_process_part(CT, _Parameters, Body, State)
+  when CT =:= <<"image/tiff">> orelse CT =:= <<"application/pdf">> ->
+    process_part(CT, Body, State);
+maybe_process_part(<<"image/", _/binary>> = CT, _Parameters, Body, State) ->
+    case whapps_config:get_binary(?CONFIG_CAT, <<"image_min_size">>) of
```

Comment:
add here a check to list of allowed images.
image_min_size should come after or andalso